src/trio/_core/_run.py

```diff
@@ -1194,6 +1232,7 @@ def parent_task(self) -> Task:
 
     def _add_exc(self, exc: BaseException) -> None:
         self._pending_excs.append(exc)
+        # TODO: source/reason?
         self.cancel_scope.cancel()
```

Comment:
I think this is adequately handled below; but I'd add a comment pointing to `_child_finished` to say so.

Reply:
ah, it isn't. Also need to handle `_nested_child_finished`, if the code inside the context manager raises.